rviz_yaml_cpp_vendor/rviz_yaml_cpp_vendor-extras.cmake.in

```diff
@@ -1,16 +1,19 @@
-find_package(yaml-cpp QUIET)
-
-if(NOT yaml-cpp_FOUND)
+if(@rviz_yaml_cpp_vendor_LOCAL_YAML@)
+  message(STATUS "Using local yaml-cpp")
   # add the local Modules directory to the modules path
   if(WIN32)
     set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/rviz_yaml_cpp_vendor/CMake")
   else()
     set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/rviz_yaml_cpp_vendor/lib/cmake/yaml-cpp")
   endif()
   message(STATUS "Setting yaml-cpp_DIR to: '${yaml-cpp_DIR}'")
+else()
+  message(STATUS "Using system yaml-cpp")
 endif()
 
-find_package(yaml-cpp CONFIG REQUIRED QUIET)
-
+find_package(yaml-cpp CONFIG REQUIRED)
 set(rviz_yaml_cpp_vendor_LIBRARIES ${YAML_CPP_LIBRARIES})
-set(rviz_yaml_cpp_vendor_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
+
+if(@rviz_yaml_cpp_vendor_LOCAL_YAML@)
+  set(rviz_yaml_cpp_vendor_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
```

Comment:
You can't make this assumption. If I install yaml cpp to `~/local` and add that to my `CMAKE_PREFIX_PATH` this assumption won't hold.

Also, it doesn't buy you anything to special case this.

Reply:
This in fact has fixed the `fatal error: stdlib.h: No such file or directory` error - if `/usr/include`, which system-wise installed yaml-cpp reports in `YAML_CPP_INCLUDE_DIR`, is passed to compiler as `-isystem` it breaks everything.

What would you suggest then?

Reply:
I would suggest that you have a special case (this version of yaml-cpp on Linux for example) and in that case iterate over the headers and remove/change the `/usr/include` instance. Assuming the include directory is not needed when we don't build it locally isn't a safe assumption to make.

> is passed to compiler as -isystem it breaks everything.

Separate question, why is the content of `YAML_CPP_INCLUDE_DIR` getting passed after `-isystem` rather than `-I`? Is something using `include_directories(SYSTEM ...)` somewhere?